WalletWasabi/Wallets/CachedBlockProvider.cs

```diff
@@ -32,7 +33,8 @@ public async Task<Block> GetBlockAsync(uint256 hash, CancellationToken cancellat
 		if (block is null)
 		{
 			block = await BlockSourceProvider.GetBlockAsync(hash, cancellationToken).ConfigureAwait(false);
-			await BlockRepository.SaveAsync(block, cancellationToken).ConfigureAwait(false);
+			await LastSaveBlockTask;
+			LastSaveBlockTask = BlockRepository.SaveAsync(block, cancellationToken).ConfigureAwait(false);
```

Comment:
This results in error, did you mean this? 
```suggestion
			LastSaveBlockTask = BlockRepository.SaveAsync(block, cancellationToken);
```

Reply:
sry I realized just before pushing that I forgot `.ConfigureAwait(false)` and I added it in the wrong line...
Ty for the test